src/worker.cpp

```diff
@@ -1185,4 +1183,300 @@ void Worker::collectFrusFromJson()
+            constants::VpdBackupMap.end())
+        {
+            return constants::VpdBackupMap.at(l_backupMap);
+        }
+    }
+    return std::monostate{};
+}
+
+std::string Worker::getHwPath(const std::string& i_pathName)
+{
+    if (i_pathName.compare("primary") == constants::STR_CMP_SUCCESS &&
+        m_parsedJson.contains("backupAndRestore"))
+    {
+        return m_parsedJson["backupAndRestore"].value("PrimaryHardwarePath",
+                                                      "");
+    }
+    else if (i_pathName.compare("secondary") == constants::STR_CMP_SUCCESS &&
+             m_parsedJson.contains("backupAndRestore"))
+    {
+        return m_parsedJson["backupAndRestore"].value("SecondaryHardwarePath",
+                                                      "");
+    }
+    return std::string{};
+}
+
+types::BinaryVector Worker::getIpzKeywordValue(
+    const types::IPZVpdMap& i_parsedVpdMap, const std::string& i_inventoryPath,
+    const std::string& i_record, const std::string& i_keyword)
+{
+    types::BinaryVector l_kwdValue;
+    if (!i_parsedVpdMap.empty())
+    {
+        if (i_parsedVpdMap.find(i_record) != i_parsedVpdMap.end())
+        {
+            std::string l_kwdStrValue;
+            utils::getKwVal(i_parsedVpdMap.at(i_record), i_keyword,
+                            l_kwdStrValue);
+            l_kwdValue = types::BinaryVector(l_kwdStrValue.begin(),
+                                             l_kwdStrValue.end());
+        }
+    }
+    else
+    {
+        // Read keyword value from DBus
+        // Uncomment when readKeyword API implemenation is done
+        (void)i_inventoryPath;
+        /* l_kwdValue = utils::readKeyword(i_inventoryPath,
+         * std::tuple<types::Record, types::Keyword>(i_record, i_keyword));*/
+    }
+    return l_kwdValue;
+}
+
+std::string Worker::getInvPath(const std::string& i_vpdFilePath,
+                               const std::string& i_pathName)
+{
+    std::string l_inventoryPath;
+    if (!i_pathName.empty())
+    {
+        if (!i_vpdFilePath.empty())
+        {
+            l_inventoryPath = utils::getInventoryPath(m_parsedJson,
+                                                      i_vpdFilePath);
+        }
+        else
+        {
+            std::string l_vpdFilePath;
+            if (i_pathName.compare("primary") == constants::STR_CMP_SUCCESS)
+            {
+                l_vpdFilePath = getHwPath("secondary");
+            }
+            else if (i_pathName.compare("secondary") ==
+                     constants::STR_CMP_SUCCESS)
+            {
+                l_vpdFilePath = getHwPath("primary");
+            }
+            else
+            {
+                return std::string{};
+            }
+
+            l_inventoryPath = utils::getInventoryPath(m_parsedJson,
+                                                      l_vpdFilePath);
+        }
+    }
+    return l_inventoryPath;
+}
+
+void Worker::backupAndRestore(types::VPDMapVariant& io_parsedVpdMap)
```

Comment:
The implementation looks to be incorrect.
Please revisit, why we need three maps when involved entity is only two. 
In case there is no requirement, please correct the implementation.

Reply:
Updated the API with only primary and secondary VPD maps.